src/main/java/ui/UI.java

```diff
@@ -540,4 +544,27 @@ public boolean isWindowMinimized() {
     public boolean isWindowFocused() {
         return mainStage.isFocused();
     }
+
+    /**
+     * Warns user if the Java runtime version is lower than HT's requirement
+     */
+    private void warnIfJavaVersionOutdated() {
+        try {
+            JavaVersion runtimeVersion = JavaVersion.fromString(System.getProperty("java.runtime.version"));
+            JavaVersion requiredVersion = JavaVersion.fromString(REQUIRED_VERSION);
+
+            if (JavaVersion.isJavaVersionTooLow(runtimeVersion, requiredVersion)) {
+                showJavaVersionOutdatedWarning(runtimeVersion, requiredVersion);
+            }
+        } catch (IllegalArgumentException e) {
+            logger.error("Java Version string failed to be parsed.", e);
```

Comment:
I will rework this. So:
- `requiredVersion` will be covered by a test case, hence there should be no way that it fails parsing. I will just log an error saying `should have been covered by test case`.
- if `runtimeVersion` cannot be parsed, will show an error dialog saying that the Java version is not compatible with HubTurbo.
- Otherwise, show the current warning.